backend/kernelCI_app/views/hardwareDetailsView.py

```diff
@@ -0,0 +1,162 @@
+from collections import defaultdict
+from datetime import timedelta
+from django.http import JsonResponse
+from django.views import View
+from django.db.models import F
+from django.utils import timezone
+from kernelCI_app.viewCommon import create_details_summary, get_details_issue
+from kernelCI_app.models import Tests
+
+
+from kernelCI_app.utils import (
+    convert_issues_dict_to_list,
+    extract_error_message,
+    extract_platform,
+)
+
+
+def get_build(record):
+    return {
+        "id": record["build_id"],
+        "architecture": record["build_architecture"],
+        "config_name": record["build_config_name"],
+        "misc": record["build_misc"],
+        "config_url": record["build_config_url"],
+        "compiler": record["build_compiler"],
+        "valid": record["build_valid"],
+        "duration": record["build_duration"],
+        "log_url": record["build_log_url"],
+        "start_time": record["build_start_time"],
+        "git_repository_url": record["checkout_git_repository_url"],
+        "git_repository_branch": record["checkout_git_repository_branch"],
+    }
+
+
+def get_history(record):
+    return {
+        "id": record["id"],
+        "status": record["status"],
+        "path": record["path"],
+        "duration": record["duration"],
+        "startTime": record["start_time"],
+        "hardware": record["environment_compatible"]
+    }
+
+
+def update_issues(issues, issue):
+    existing_issue = issues.get(issue["id"])
+    if existing_issue:
+        existing_issue["incidents_info"]["incidentsCount"] += 1
+    else:
+        issues[issue["id"]] = issue
+
+
+def isBoot(record):
+    return record["path"] == "boot" or record["path"].startswith("boot.")
+
+
+class HardwareDetails(View):
+    def sanitize_records(self, records):
+        processedBuilds = set()
+        builds = {"items": [], "issues": {}}
+
+        tests = {
+            "history": [],
+            "platformsFailing": set(),
+            "statusSummary": defaultdict(int),
+            "failReasons": defaultdict(int),
+            "configs": defaultdict(lambda: defaultdict(int)),
+            "hardware": defaultdict(lambda: defaultdict(int)),
+            "issues": {}
+        }
+
+        boots = {
+            "history": [],
+            "platformsFailing": set(),
+            "statusSummary": defaultdict(int),
+            "failReasons": defaultdict(int),
+            "configs": defaultdict(lambda: defaultdict(int)),
+            "hardware": defaultdict(lambda: defaultdict(int)),
+            "issues": {}
+        }
+
+        for r in records:
+            build_id = r["build_id"]
+            issue_id = r["issue_id"]
+            status = r["status"]
+
+            tests_or_boots = boots if isBoot(r) else tests
```

Comment:
here is pointing to a variable in an upper scope, there is a big change of modifying instead of overriding because it is passing by reference since it is iterating thousands of time